### ISSUE TITLE
Serde's README incomplete on crates.io

### ISSUE THREAD

**Issue by mbernat** [rust-lang/crates.io#2270]
If you compare https://crates.io/crates/serde with the original https://github.com/serde-rs/serde you can see that the former lacks Cargo.toml part of the usage instructions. This is unfortunate since following these broken instructions leads to unnecessary errors.

The missing part is contained inside `<details>...<details>` tags. Best guess is this is being stripped out by some markdown processor or something.

**Comment by locks** [comment]
Yeah, the markdown processor has an allowlist API that allows for more control. People do… interesting things in their READMEs.

**Comment by mbernat** [comment]
Right. Still, it would be nice to fix it so that new people can start easily, since it's such a popular crate.

What's the best way forward? Allowing more stuff in the processor or convincing serde maintainers to update the README? Although having a `<details>` tag in there doesn't strike me as particularly crazy, so I'm voting for the first option :)

**Comment by locks** [comment]
Yeah, the team has to do a lil' analyses of the impact of allowing `<details>` and then make the call. If the decision is negative then updating the README would be the next alternative 😅.

**Comment by carols10cents** [comment]
[We mostly use ammonia's defaults](https://github.com/rust-lang/crates.io/blob/125aff9c6020291b1e27772dec82004d463f533a/src/render.rs#L46-L55), which [allows the `details` tag](https://github.com/rust-ammonia/ammonia/blob/892da8a7f16b7af34c95394b98b5d0ca974ae9cd/src/lib.rs#L291), so that's not the problem. I'm going to investigate more.

**Comment by carols10cents** [comment]
Crates.io uses the file specified in a crate's `Cargo.toml` `readme` metadata, and because serde is a workspace, [the serde crate specifies](https://github.com/serde-rs/serde/blob/1894cb703f480e9b57f7515599a7c3377b45567b/serde/Cargo.toml#L12) a file named [crates-io.md](https://github.com/serde-rs/serde/blob/1894cb703f480e9b57f7515599a7c3377b45567b/crates-io.md), which is not the same as the workspace repo's README.md and [that might be intentional](https://github.com/serde-rs/serde/commit/c50c9d88625ab15327cc07e5505432553087287e#diff-bd90a158d620a922a396decb5f65ba93).

TL;DR This is an issue with serde, not crates.io.

**Comment by locks** [comment]
Oh, nice solve! Thanks for looking into it :)

**Comment by mbernat** [comment]
Thank you very much! I'll try talking to serde people then.
